code/modules/surgery/organs/heart.dm

```diff
@@ -254,3 +254,192 @@
+		if(CRYSTALIZE_STAGE_SMALL to INFINITY)
+			examine_list += "<span class='notice'>Some crystals are coming out of [examined_human] </span>"
+
+///On stat changes, if the victim is no longer dead but theyre crystalizing, cancel it, if they become dead, start the crystalizing process if possible
+/obj/item/organ/heart/ethereal/proc/on_stat_change(mob/living/victim, new_stat)
+	SIGNAL_HANDLER
+
+	if(new_stat != DEAD)
+		if(crystalize_timer_id)
+			stop_crystalization_process(victim)
+		return
+
+
+	if(QDELETED(victim) || victim.suiciding)
+		return //lol rip
+
+	if(!COOLDOWN_FINISHED(src, crystalize_cooldown))
+		return //lol double rip
+
+	to_chat(victim, "<span class='nicegreen'>Crystals start forming around your dead body</span>")
+	victim.visible_message("<span class='notice'>Crystals start forming around [victim]</span>")
+	ADD_TRAIT(victim, TRAIT_CORPSELOCKED, SPECIES_TRAIT)
+
+	crystalize_timer_id = addtimer(CALLBACK(src, .proc/crystalize, victim), CRYSTALIZE_PRE_WAIT_TIME, TIMER_STOPPABLE)
+
+	RegisterSignal(victim, COMSIG_HUMAN_DISARM_HIT, .proc/reset_crystalizing)
+	RegisterSignal(victim, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+	RegisterSignal(victim, COMSIG_MOB_APPLY_DAMGE, .proc/on_take_damage)
+
+///Ran when disarmed, prevents the ethereal from reviving
+/obj/item/organ/heart/ethereal/proc/reset_crystalizing(mob/living/defender, mob/living/attacker, zone)
+	SIGNAL_HANDLER
+	to_chat(defender, "<span class='notice'>The crystals on your corpse are gently broken off, and will need some time to recover.</span>")
+	defender.visible_message("<span class='notice'>The crystals on [defender] are gently broken off</span>")
+	deltimer(crystalize_timer_id)
+	crystalize_timer_id = addtimer(CALLBACK(src, .proc/crystalize, defender), CRYSTALIZE_DISARM_WAIT_TIME, TIMER_STOPPABLE) //Lets us restart the timer on disarm
+
+
+///Actually spawns the crystal which puts the ethereal in it.
+/obj/item/organ/heart/ethereal/proc/crystalize(mob/living/ethereal)
+	if(!COOLDOWN_FINISHED(src, crystalize_cooldown) || ethereal.stat != DEAD)
+		return //Should probably not happen, but lets be safe.
+	COOLDOWN_START(src, crystalize_cooldown, INFINITY) //Prevent cheeky double-healing until we get out, this is against stupid admemery
+	current_crystal = new(get_turf(ethereal), src)
+	stop_crystalization_process(ethereal)
+
+///Stop the crystalization process, unregistering any signals and resetting any variables.
+/obj/item/organ/heart/ethereal/proc/stop_crystalization_process(mob/living/ethereal)
+	UnregisterSignal(ethereal, COMSIG_HUMAN_DISARM_HIT)
+	UnregisterSignal(ethereal, COMSIG_PARENT_EXAMINE)
+	UnregisterSignal(ethereal, COMSIG_MOB_APPLY_DAMGE)
+
+	crystalization_process_damage = 0 //Reset damage taken during crystalization
+
+	if(crystalize_timer_id)
+		deltimer(crystalize_timer_id)
+		crystalize_timer_id = null
+
+
+///Lets you stop the process with enough brute damage
```

Comment:
are vacuums and such intended to stop ethereal self-rezzing? it seems like this is intended to revive you from death to environmental hazards, but currently, it won't revive you from death due to being spaced and such.

Reply:
does space do brute dmg?

Reply:
Yes

Reply:
i'll look into a way to prevent it

Reply:
Any progress on this?

Reply:
I tested it, space does not trigger this, (But beating them up with a toolbox will)